core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -751,6 +739,18 @@ abstract Query rangeQuery(String field, Object lowerTerm, Object upperTerm,
         Number valueForSearch(Number value) {
             return value;
         }
+
+        /**
+         * Returns true if the object is a number and has a decimal part
+         */
+        protected boolean hasDecimalPart(Object number) {
+            if (number instanceof Number) {
+                double doubleValue = ((Number) number).doubleValue();
+                return doubleValue % 1 != 0;
+            }
+            return false;
```

Comment:
Is it safe to return false otherwise? Maybe `return Double.parseDouble(number) % 1 != 0`?